### PR TITLE
resize interactives

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?
The bottom of interactives can be cut off. They are currently set to 500px in height.

## Screenshots

| Before | After |
| --- | --- |
|  |  |

### CODE REVIEW

File: src/client/article.ts [hunk 1]
```diff
@@ -214,6 +214,45 @@ function hasSeenCards(): void {
     })
 }
 
+function safeParse(item: {} | string): any {
```
Comment (JamieB-gu) [inline src/client/article.ts:217]: It looks like most of the validation errors can be traced back to this `any` - I think we can get around this by changing it to `unknown`?

File: src/client/article.ts [hunk 2]
```diff
@@ -214,6 +214,45 @@ function hasSeenCards(): void {
...
+
+            switch (message.type) {
+                case 'set-height':
+                    iframe.height = message.value;
+                    break;
+                default:
+            }
+        }, false);
+
+        figure.replaceChild(iframe, serverRenderedIframe);
```
Comment (JamieB-gu) [inline src/client/article.ts:252]: What made you opt to replace the server-rendered `iframe` instead of attaching listeners for messages from it?

**Review by JamieB-gu — COMMENTED** [review]
Looks good, couple comments.

File: src/client/article.ts (hunk 2 above)
Comment (webb04) [reply src/client/article.ts:252]: The parent window receives messages via `postMessage` in the iframe to `set-height` but I couldn't work out how to match that request to a particular iframe.

We have the `origin` available in every request to update the height, but we would have a problem if we have two interactives from the same origin. The only solution I could find was to replace them all client side - this is how the templates currently handle them.

File: src/client/article.ts (hunk 1 above)
Comment (webb04) [reply src/client/article.ts]: Updated this PR, I thought we had handled this before...

**Review by JamieB-gu — APPROVED** [review]
Nice! 🎉

**Pull request merged**